### PR TITLE
Refactor #9110: Use a vector for station locations list in sub_6CB945

### PR DESCRIPTION
**Pull request by aaruel**
#9110

### CODE REVIEW

File: src/openrct2/ride/Ride.cpp [hunk 1]
```diff
@@ -7114,42 +7114,38 @@ void sub_6CB945(Ride* ride)
...
-            *locationList++ = exit;
+            locations.push_back(exit);
             ride_clear_exit_location(ride, stationId);
         }
     }
-    (*locationList++).x = COORDS_NULL;
 
-    locationList = locations;
-    for (; !(*locationList).isNull(); locationList++)
+    for (auto locationList = locations.cbegin(); locationList != locations.cend(); locationList++)
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Ride.cpp]: This doesn't look pretty a ranged for loop would work here.

File: src/openrct2/ride/Ride.cpp [hunk 2]
```diff
@@ -7114,42 +7114,38 @@ void sub_6CB945(Ride* ride)
...
     {
-        TileCoordsXYZD* locationList2 = locationList;
+        auto locationList2 = locationList;
         locationList2++;
 
         bool duplicateLocation = false;
-        do
+        for (; locationList2 != locations.cend(); locationList2++)
```
[the remark this answers is not in the document]
Comment (aaruel) [reply src/openrct2/ride/Ride.cpp]: Agreed, which is why I didn't go for the ranged for loop above, figuring having access to the iterator would keep finding duplicates simple. I can think up a prettier solution if you wanted

File: src/openrct2/ride/Ride.cpp [hunk 3]
```diff
@@ -7114,49 +7114,46 @@ void sub_6CB945(Ride* ride)
...
+            locations.push_back(exit);
             ride_clear_exit_location(ride, stationId);
         }
     }
-    (*locationList++).x = COORDS_NULL;
 
-    locationList = locations;
-    for (; !(*locationList).isNull(); locationList++)
+    auto locationListIter = locations.cbegin();
+    for (const TileCoordsXYZD& locationCoords : locations)
```
Comment (aaruel) [inline src/openrct2/ride/Ride.cpp]: Should I prefer `const auto&` here and on L7143?

**Comment by aaruel** [comment]
@duncanspumpkin Thank you! Yes I tested and it works as intended.

**Pull request merged**